RelayTestApp/Assets/Scripts/GameManager/GameManager.cs

```diff
@@ -0,0 +1,268 @@
+        if (_userCursorsList.Count > 0)
+        {
+            for (int i = _userCursorsList.Count - 1; i > -1; i--)
+            {
+                Destroy(_userCursorsList[i].gameObject);
+            }
+            _userCursorsList.Clear();
+        }
+        Color newColor;
+        for (int i = 0; i < lobby.Members.Count; i++)
+        {
+            UserCursor newCursor = Instantiate(UserCursorPrefab, Vector3.zero, Quaternion.identity, UserCursorParent.transform);
+            
+            newCursor.AdjustVisibility(false);
+            newColor = ReturnUserColor(lobby.Members[i].UserGameColor);
+            newCursor.SetUpCursor(newColor,lobby.Members[i].Username);
+            lobby.Members[i].UserCursor = newCursor;
+            _userCursorsList.Add(newCursor);
+            if (lobby.Members[i].Username == CurrentUserInfo.Username)
+            {
+                GameArea.LocalUserCursor = newCursor;
+            }
+        }
+    }
+    public void UpdateLobbyState()
+    {   
+        AdjustEntryList(UserEntryLobbyParent.transform,UserEntryLobbyPrefab);
+        StartGameBtn.SetActive(IsLocalUserHost());
+    }
+    
+    /// <summary>
+    /// After list of users is generated for the current match, call this to display the connected users
+    /// </summary>
+    public void UpdateMatchState()
+    {
+        AdjustEntryList(UserEntryMatchParent.transform,UserEntryMatchPrefab);
+    }
+
+    private void AdjustEntryList(Transform parent, UserEntry prefab)
+    {
+        if (_matchEntries.Count > 0)
+        {
+            for(int i = _matchEntries.Count - 1; i > -1; i--)
+            {
+                Destroy(_matchEntries[i].gameObject);
+            }
+            _matchEntries.Clear();    
+        }
+        
+        //populate user entries based on members in lobby
+        Lobby lobby = StateManager.Instance.CurrentLobby;
+        for (int i = 0; i < lobby.Members.Count; i++)
+        {
+            var newEntry = Instantiate(prefab, Vector3.zero, Quaternion.identity,parent);
+            SetUpUserEntry(lobby.Members[i], newEntry);
+            _matchEntries.Add(newEntry);
+        }
+    }
+    
+    private void SetUpUserEntry(UserInfo info,UserEntry entry)
+    {
+        entry.UsernameText.text = info.Username;
+        Color userColor = ReturnUserColor(info.UserGameColor);
+        entry.UsernameText.color = userColor;
+        if (entry.UserDotImage != null)
+        {
+            entry.UserDotImage.color = userColor;
+        }
+    }
+
+    public void AdjustUserShockwaveMask(string username,bool isVisible)
+    {
+        //populate user entries based on members in lobby
+        Lobby lobby = StateManager.Instance.CurrentLobby;
+        for (int i = 0; i < lobby.Members.Count; i++)
+        {
+            if (lobby.Members[i].Username.Equals(username))
+            {
+                lobby.Members[i].AllowSendTo = isVisible;
+            }
+            else if (CurrentUserInfo.Username.Equals(username))
```

Comment:
you check for current user for each member? Probably do that check outside the for loop